### ISSUE TITLE
RingAlgorithm does not derive `Debug`

### ISSUE THREAD

**Issue by GlenDC** [rustls/webpki#213]
https://github.com/rustls/webpki/blob/6a8ae20089d78c4d39d50ff65ca48fa83794386b/src/ring_algs.rs#L21C8-L21C21 (`RingAlgorithm`) does not implement the `Debug` trait.

This is an issue as https://github.com/rustls/pki-types/blob/d99520cbdca7c1b55f6008a7a6a03f6af4e8b1cd/src/lib.rs#L323 requires this trait, making it in the newest alpha versions impossible to use `RingAlgorithm` where this trait is expected.

As a result I have currently the current error:

```
error[E0277]: `RingAlgorithm` doesn't implement `Debug`
   --> /Users/glendc/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-webpki-0.102.0-alpha.7/src/ring_algs.rs:27:41
    |
27  | impl SignatureVerificationAlgorithm for RingAlgorithm {
    |                                         ^^^^^^^^^^^^^ `RingAlgorithm` cannot be formatted using `{:?}`
    |
    = help: the trait `Debug` is not implemented for `RingAlgorithm`
    = note: add `#[derive(Debug)]` to `RingAlgorithm` or manually `impl Debug for RingAlgorithm`
note: required by a bound in `SignatureVerificationAlgorithm`
   --> /Users/glendc/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-pki-types-0.2.3/src/lib.rs:323:57
    |
323 | pub trait SignatureVerificationAlgorithm: Send + Sync + fmt::Debug {
    |                                                         ^^^^^^^^^^ required by this bound in `SignatureVerificationAlgorithm`
help: consider annotating `RingAlgorithm` with `#[derive(Debug)]`
    |
21  + #[derive(Debug)]
22  | struct RingAlgorithm {
```

**Comment by cpu** [comment]
Thanks for opening an issue. This was my bad - I accidentally broke semver with pki-types 0.2.3 not realizing this debug bound was added to the trait. I already have a fix for webpki up for review: https://github.com/rustls/webpki/pull/212

**Comment by GlenDC** [comment]
Thank you. I approved it. But that's probably not helping anything for the progress of that PR 😓

**Comment by cpu** [comment]
This is fixed in [v0.102.0-alpha.8](https://github.com/rustls/webpki/releases/tag/v%2F0.102.0-alpha.8). Apologies for the breakage!
